include/EASTL/map.h

```diff
@@ -439,6 +443,33 @@ namespace eastl
 	#endif
 
 
+	template <typename Key, typename T, typename Compare, typename Allocator>
+	inline T& map<Key, T, Compare, Allocator>::at(const Key& key)
+	{
+		iterator itLower(lower_bound(key)); // itLower->first is >= key.
+
+		if(itLower == end())
+		{
+			throw std::out_of_range("map::at key does not exist");
```

Comment:
EASTL can't assume that exceptions are enabled.  The vast majority of the time we disable exceptions so we typical add asserts in the case when exceptions are disabled.  You can see examples of how this done through the library but most recently I implemented string_view with similar code.

https://github.com/electronicarts/EASTL/blob/master/include/EASTL/string_view.h#L89